src/utilities/coerceInputValue.js

```diff
@@ -0,0 +1,209 @@
+// @flow strict
+
+import { forEach, isCollection } from 'iterall';
+
+import objectValues from '../polyfills/objectValues';
+
+import inspect from '../jsutils/inspect';
+import didYouMean from '../jsutils/didYouMean';
+import isObjectLike from '../jsutils/isObjectLike';
+import suggestionList from '../jsutils/suggestionList';
+import printPathArray from '../jsutils/printPathArray';
+import { type Path, addPath, pathToArray } from '../jsutils/Path';
+
+import { GraphQLError } from '../error/GraphQLError';
+import {
+  type GraphQLInputType,
+  isScalarType,
+  isEnumType,
+  isInputObjectType,
+  isListType,
+  isNonNullType,
+} from '../type/definition';
+
+type OnErrorCB = (
+  path: $ReadOnlyArray<string | number>,
+  invalidValue: mixed,
+  error: GraphQLError,
+) => void;
+
+/**
+ * Coerces a JavaScript value given a GraphQL Input Type.
+ */
+export function coerceInputValue(
```

Comment:
I decided to create a new `coerceInputValue` function sine `coerceValue` is part of public API so we can't change its behavior.

In addition to that the new name is less confusing since GraphQL Spec defines `Input coercion` and `Result coercion`:
https://graphql.github.io/graphql-spec/draft/#sec-Scalars.Input-Coercion